File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        List<Integer> row = new ArrayList<>();
        List<Integer> column = new ArrayList<>();
        for(int i = 0;i<matrix.length;i++){
            for(int j = 0;j<matrix[0].length;j++){
                if(matrix[i][j] == 0){
                    row.add(i);
                    column.add(j);
                }

            }
        }
        for(int i : row){
            for(int j = 0;j<matrix[0].length;j++) matrix[i][j] = 0;
        }

        for(int i : column){
            for(int j = 0;j<matrix.length;j++) matrix[j][i] = 0;
        }
    }
}

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        ROWS, COLS = len(matrix), len(matrix[0])
        zeros = [(row,col) for row in range(ROWS) for col in range(COLS) if matrix[row][col] == 0]

        zeros = set(zeros)

        for r,c in zeros:
            matrix[r] = [0]*COLS
            for i in range(ROWS):
                matrix[i][c] = 0
        